requirements-dev.txt

```diff
@@ -19,4 +19,5 @@ cmake
 custatevec-cu12
 pylint
 scipy~=1.12.0
+autoray<0.6.10
```

Comment:
In PL we have a req `autoray>=0.6.1`, so don't we want to make it `autoray==0.6.1`?

Reply:
I think `autoray < 0.6.10` should be enough as a temporary solution which only affects developers.